src/python/pants/backend/python/mixed_interpreter_constraints/py_constraints.py

```diff
@@ -140,18 +140,6 @@ async def py_constraints(
         transitive_targets.closure, python_setup
     )
 
-    if not final_constraints:
```

Comment:
This case can no longer occur - we always get at least the global constraints.

Reply:
this was a bad change, now this happens:

```
❯ ./pants py-constraints build-support/common.sh
Final merged constraints: CPython==3.9.*

CPython==3.9.*
  build-support/common.sh
```

vs

```
❯ ./pants py-constraints build-support/common.sh
11:09:38.80 [WARN] No Python files/targets matched for the `py-constraints` goal. All target types with Python interpreter constraints: pex_binary, python_source, python_sources, python_test, python_test_utils, python_tests
```

Reply:
Yes, this is on purpose, as explained in the change.

Reply:
Fixed in #13470